src/ducks/memoRequiredAccounts.ts

```diff
@@ -0,0 +1,72 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import {
+  MEMO_REQ_ACCOUNT_STORAGE_ID,
+  MEMO_REQ_ACCOUNT_DATE_STORAGE_ID,
+} from "constants/settings";
+import { getMemoRequiredAccounts } from "helpers/getMemoRequiredAccounts";
+import {
+  ActionStatus,
+  MemoRequiredAccountsInitialState,
+  MemoRequiredAccountsResponse,
+} from "types/types.d";
+
+export const fetchMemoRequiredAccountsAction = createAsyncThunk<
+  MemoRequiredAccountsResponse | {}
+>("memoRequiredAccounts/fetchMemoRequiredAccountsAction", async () => {
+  let result;
+
+  const date = new Date();
+  const time = date.getTime();
+  const sevenDaysAgo = time - 7 * 24 * 60 * 60 * 1000;
+  const memoRequiredAccountsCacheDate = Number(
+    localStorage.getItem(MEMO_REQ_ACCOUNT_DATE_STORAGE_ID),
+  );
+
+  result = JSON.parse(
+    localStorage.getItem(MEMO_REQ_ACCOUNT_STORAGE_ID) || "[]",
+  );
+
+  // if memoRequiredAccounts were last cached over seven days ago,
+  // make the request
+  // memoRequiredAccountsCacheDate is coerced to 0 if not found in storage
+  if (memoRequiredAccountsCacheDate < sevenDaysAgo) {
+    try {
+      result = await getMemoRequiredAccounts();
+      // store the result plus the date we've acquired them
+      localStorage.setItem(MEMO_REQ_ACCOUNT_STORAGE_ID, JSON.stringify(result));
+      localStorage.setItem(MEMO_REQ_ACCOUNT_DATE_STORAGE_ID, time.toString());
+    } catch (e) {
+      console.error("Memo required accounts API did not respond");
+    }
+  }
+
+  return result;
```

Comment:
it might not be possible to do this in a succinct way, but it'd be nice if we could combine this with the logic in `fetchFlaggedAccountsAction`. Maybe a helper that accepts `MEMO_REQ_ACCOUNT_STORAGE_ID `, `MEMO_REQ_ACCOUNT_DATE_STORAGE_ID` and `getMemoRequiredAccounts`

For ex: `fetchMemoizedData({ dataStorageId, dataStorageDateId,   fetchMethod })`